backend/src/Controllers/CloudPostsController.php

```diff
@@ -0,0 +1,282 @@
+
+				}
+			}
+		} else {
+
+			$override_post['post_content'] = $post_data->post_content;
+			if ( is_array( $post_cat ) && count( $post_cat ) !== 0 ) {
+
+				foreach ( $post_cat as $category ) {
+
+					   $term = term_exists( $category->slug, $category->taxonomy );
+
+					   wp_set_post_terms( $post_id, [ $term['term_taxonomy_id'] ], $category->taxonomy, true );
+
+				}
+			}
+		}
+
+		if ( is_wp_error( wp_update_post( $override_post ) ) ) {
+
+			return rest_ensure_response(
+				[
+					'error' => true,
+				]
+			);
+		} else {
+			return rest_ensure_response(
+				[
+					'success' => true,
+				]
+			);
+		}
+
+	}
+
+
+
+	function save_to_library( $request ) {
+		$id = $request->get_param( 'id' );
+		$library_id = $request->get_param( 'library_id' );
+
+		$post = get_post( $id );
+		$client = new \FL\Assistant\Clients\Cloud\CloudClient;
+
+		$media_path = get_attached_file( get_post_thumbnail_id( $post ) );
+		$media = $media_path ? curl_file_create( $media_path ) : null;
+
+		$taxonomies = get_taxonomies( '', 'names' );
+		$post_taxonomies = wp_get_object_terms( $id, $taxonomies );
+
+		$comments = get_comments( [ 'post_id' => $id ] );
+
+		return $client->libraries->createItem(
+			$library_id,
+			[
+				'name'  => $post->post_title,
+				'type'  => 'post',
+				'data'  => [
+					'post'     => $post,
+					'meta'     => get_post_meta( $id ),
+					'terms'    => $post_taxonomies,
+					'comments' => $comments,
+				],
+				'media' => $media,
+			]
+		);
+	}
+
+
+	function import_from_library( $request ) {
+
+		global $wpdb;
+
+		$item_id = $request->get_param( 'item_id' );
+		$client = new \FL\Assistant\Clients\Cloud\CloudClient;
+		$response = $client->libraries->getItem( $item_id );
+		$post_data = $response->data->post;
+		$post_meta = $response->data->meta;
+		$post_cat  = $response->data->terms;
+		$current_user = wp_get_current_user();
+
+		$newpost_data = [
+			'post_author'  => $current_user->ID,
+			'post_content' => $post_data->post_content ? $post_data->post_content : '',
+			'post_excerpt' => $post_data->post_excerpt ? $post_data->post_excerpt : '',
+			'post_name'    => $post_data->post_name ? $post_data->post_name : '',
+			'post_status'  => 'draft',
+			'post_title'   => $post_data->post_title,
+			'post_type'    => $post_data->post_type,
+			'to_ping'      => $post_data->ping_status,
+			'menu_order'   => $post_data->menu_order,
+		];
+
+		$new_post_id = wp_insert_post( $newpost_data );
+
+		if ( is_wp_error( $new_post_id ) ) {
+
+			return rest_ensure_response(
+				[
+					'error' => true,
+				]
+			);
+
+		} else {
+
+			wp_set_post_terms( $new_post_id, null, 'category' );
+
+			if ( $post_meta && count( $post_meta ) !== 0 ) {
+
+				foreach ( $post_meta as $key => $value ) {
+
+					$meta_key   = $key ? $key : '';
+					$meta_value = $value[0] ? addslashes( $value[0] ) : '';
+					if ( '' !== $meta_key && '' !== $meta_value ) {
```

Comment:
I am thinking that the data are stored on the cloud so may be possible at some time or in such case data will not properly store so that's why I put condition.